x/bridge/keeper/pause.go

```diff
@@ -0,0 +1,58 @@
+package keeper
+
+import (
+	"fmt"
+
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/mezo-org/mezod/x/bridge/types"
+	evmtypes "github.com/mezo-org/mezod/x/evm/types"
+)
+
+// GetPauser returns the current pauser address.
+func (k Keeper) GetPauser(ctx sdk.Context) sdk.AccAddress {
+	store := ctx.KVStore(k.storeKey)
+
+	pauser := store.Get(types.PauserKey)
+
+	if len(pauser) == 0 {
+		pauser = evmtypes.HexAddressToBytes(evmtypes.ZeroHexAddress())
+	}
+
+	return pauser
+}
+
+// SetPauser sets the pauser address.
+func (k Keeper) SetPauser(ctx sdk.Context, pauser sdk.AccAddress) {
+	store := ctx.KVStore(k.storeKey)
+
+	if len(pauser) == 0 {
+		pauser = evmtypes.HexAddressToBytes(evmtypes.ZeroHexAddress())
+	}
+
+	store.Set(types.PauserKey, pauser)
+}
+
+// PauseBridgeOut sets the outflow limit to 0 for all supported tokens.
+func (k Keeper) PauseBridgeOut(ctx sdk.Context, caller sdk.AccAddress) error {
+	pauser := k.GetPauser(ctx)
+	if evmtypes.IsZeroHexAddress(evmtypes.BytesToHexAddress(pauser)) {
+		return fmt.Errorf("no pauser is set")
+	}
+
+	if !pauser.Equals(caller) {
+		return fmt.Errorf("caller is not the pauser")
+	}
+
+	// Set outflow limit to 0 for BTC token
+	btcToken := evmtypes.HexAddressToBytes(evmtypes.BTCTokenPrecompileAddress)
+	k.SetOutflowLimit(ctx, btcToken, math.ZeroInt())
```

Comment:
Reversing the pause would require iterating over all tokens and re-setting outflow limits higher.
Wouldn't having a single bool flag in the Bridge keeper be simpler?
We could easily pause/unpause/check if paused.
(Just a suggestion, the current implementation of pausing will work and ideally pausing shouldn't happen at all).